lib/katello/tasks/clean_installed_packages.rake

```diff
@@ -0,0 +1,52 @@
+namespace :katello do
+  desc "Clean duplicate installed packages"
+  task :clean_installed_packages => :environment do
+    User.current = User.anonymous_admin
+
+    puts "Populating cache"
+    cache = PackageCache.new
+    cache.populate
+
+    total = ::Host.count
+    count = 1
+    ::Host.find_each do |host|
+      puts "Host #{count}/#{total}"
+
+      if host.installed_package_ids.any?
+        nvras = host.installed_packages.pluck(:nvra)
+        ids = nvras.map { |nvra| cache.fetch(nvra) }
+        existing_ids = host.installed_package_ids.to_a
+        unless existing_ids.sort == ids.sort
+          host.sync_package_associations(ids)
+        end
+      end
+      count += 1
+    end
```

Comment:
I'm not clear on what this loop does. What is the advantage of this vs just calling sync_package_associations on each host and leting the optimizations in that method handle things?